Support C++ statically_known_true

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #151348
* #151347
* __->__ #151346

Differential Revision: [D73040543](https://our.internmc.facebook.com/intern/diff/D73040543/)

cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv

tugsbayasgalan committed:
Support C++ statically_known_true



Differential Revision: [D73040543](https://our.internmc.facebook.com/intern/diff/D73040543/)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Support C++ statically_known_true"


Differential Revision: [D73040543](https://our.internmc.facebook.com/intern/diff/D73040543/)

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Support C++ statically_known_true"


Differential Revision: [D73040543](https://our.internmc.facebook.com/intern/diff/D73040543/)

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Support C++ statically_known_true"


Differential Revision: [D73040543](https://our.internmc.facebook.com/intern/diff/D73040543/)

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka commented on c10/core/SymBool.h:
```diff
@@ -1,3 +1,4 @@
+
```
?

laithsakka commented on test/export/test_export.py:
```diff
@@ -696,6 +696,31 @@ def forward(self, x):
+            dynamic_shapes=dynamic_shapes,
+            strict=False,
+        )
+        print(ep.graph)
```
do you want this print?

laithsakka reviewed (approved):
please adress my comments before landing

tugsbayasgalan committed:
Update on "Support C++ statically_known_true"


Differential Revision: [D73040543](https://our.internmc.facebook.com/intern/diff/D73040543/)

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]
